src/execution/vm/vm.cpp

```diff
@@ -2112,9 +2112,9 @@ void VM::Interpret(const uint8_t *ip, Frame *frame) {  // NOLINT
   OP(Concat) : {
     auto *result = frame->LocalAt<sql::StringVal *>(READ_LOCAL_ID());
     auto *exec_ctx = frame->LocalAt<exec::ExecutionContext *>(READ_LOCAL_ID());
-    auto *left = frame->LocalAt<const sql::StringVal *>(READ_LOCAL_ID());
-    auto *right = frame->LocalAt<const sql::StringVal *>(READ_LOCAL_ID());
-    OpConcat(result, exec_ctx, left, right);
+    auto *num_inputs = frame->LocalAt<const int64_t *>(READ_LOCAL_ID());
+    auto **inputs = frame->LocalAt<const sql::StringVal **>(READ_LOCAL_ID());
+    OpConcat(result, exec_ctx, *num_inputs, *inputs);
```

Comment:
Change to (1) use an immediate for the input count, and (2) use a proper array of pointers for the inputs.

Reply:
Updated (2), waiting on input for (1) from other question.